### DISCUSSION TITLE
Control multiple CCPotentiometers

### DISCUSSION THREAD

**Discussion by hatoribe** [tttapa/Control-Surface#743] [Q&A] [answered]
I am trying to control a knob with 4 analog pins.

This is all in my code.

```
// Include the library
#include <Control_Surface.h>

// Instantiate a MIDI Interface to use
USBMIDI_Interface midi;

NoteButton buttons[] = {
  {8, {60, CHANNEL_1}},
  {9, {61, CHANNEL_1}},
  {10, {62, CHANNEL_1}},
  {11, {63, CHANNEL_1}},
};

CCPotentiometer potentiometer[] = {
{A0, {71, CHANNEL_1}},
{A1, {72, CHANNEL_1}},
{A2, {73, CHANNEL_1}},
{A3, {74, CHANNEL_1}},
};

// Initialize the Control Surface
void setup() {
  Control_Surface.begin();
}

// Update the Control Surface
void loop() {
  Control_Surface.loop();
}
```

The problem is that I'm having trouble seeing that the CC I set for a pin is applied to a single knob.
As an example, if I move the A0 knob, I want only CC71 on ch1 to react, but all cc's from 71~74 will move.
Is there any way to address this?

https://tttapa.github.io/Control-Surface-doc/Doxygen/d3/d8d/Multiple-Control-Change-Potentiometers_8ino-example.html
I have tried the code you have here, but I get the same behavior.

**Comment by tttapa** [comment, marked as answer]
Sounds like you have cross-talk between ADC channels. What's the resistance of the potentiometers you're using? Ideally, you'd use 5-10kΩ pots.

**Reply by hatoribe** [reply]
Thank you for your response.

I am using a B10KΩ variable resistor...

By the way, is it safe to recognize that this is correct for the code itself?

**Reply by hatoribe** [reply]
We have solved this problem.
The Arduino documentation says the following, which seems to be the cause of the problem.

> Performing analogRead on an unconnected pin will result in unstable and fluctuating values. This is due to a variety of factors, and even just moving your hand close to the pin will cause the value to change.

I was still in the testing phase so I only had one pin connected.
Your answer led me to this.
Thank you.
